### PR TITLE
Build JS service leak detection on top of FinalizationRegistry

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
### ⚠️ Stacked PR !

File: zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt [hunk 1]
```diff
@@ -38,26 +41,39 @@ class LeakedServicesTest {
...
   /** Use a separate method so there's no hidden reference remaining on the stack. */
   private fun allocateAndLeakService() {
-    zipline.take<EchoService>("helloService")
+    zipline.take<EchoService>("helloService") // Deliberately not closed for testing.
+  }
+
+  /** Just attempting to take a service causes Zipline to process leaked services. */
+  private fun triggerJvmLeakDetection() {
+    try {
+      zipline.take<EchoService>("noSuchService")
```
Comment (oldergod) [inline zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt]: Is the throwing random? If we're expecting it to throw, we could follow a `fail()` method

**Review by oldergod — APPROVED** [review]

File: zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt (hunk 1 above)
Comment (JakeWharton) [reply zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt]: Should use `assertFailsWith<Exception> { ... }`

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/platform.kt [hunk 2]
```diff
@@ -21,6 +21,7 @@ internal const val ziplineInternalPrefix = "zipline/"
 internal const val passByReferencePrefix = "service/"
 internal const val eventLoopName = "${ziplineInternalPrefix}event_loop"
 internal const val consoleName = "${ziplineInternalPrefix}console"
+internal const val eventListenerName = "${ziplineInternalPrefix}/event_listener"
```
Comment (JakeWharton) [inline zipline/src/commonMain/kotlin/app/cash/zipline/internal/platform.kt]:
```suggestion
internal const val eventListenerName = "${ziplineInternalPrefix}event_listener"
```

**Review by JakeWharton — APPROVED** [review]

File: zipline/src/commonMain/kotlin/app/cash/zipline/internal/platform.kt (hunk 2 above)
Comment (swankjesse) [reply zipline/src/commonMain/kotlin/app/cash/zipline/internal/platform.kt]: Good catch!

File: zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt (hunk 1 above)
Comment (swankjesse) [reply zipline/src/jniTest/kotlin/app/cash/zipline/LeakedServicesTest.kt]: Fixed.

**Pull request merged**